components/builder-api/src/config.rs

```diff
@@ -166,19 +166,38 @@ impl GatewayCfg for Config {
 pub struct HttpCfg {
     pub listen:        IpAddr,
     pub port:          u16,
+    pub tls:           Option<TLSCfg>,
     pub handler_count: usize,
     pub keep_alive:    usize,
 }
 
+/// Optional TLS configuration
+#[derive(Clone, Debug, Deserialize)]
+#[serde(default)]
+pub struct TLSCfg {
+    pub cert_path:    PathBuf,
+    pub key_path:     PathBuf,
+    pub ca_cert_path: Option<PathBuf>,
+}
+
 impl Default for HttpCfg {
     fn default() -> Self {
         HttpCfg { listen:        IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)),
                   port:          9636,
+                  tls:           None,
                   handler_count: Config::default_handler_count(),
                   keep_alive:    60, }
     }
 }
 
+impl Default for TLSCfg {
+    fn default() -> Self {
+        TLSCfg { cert_path:    PathBuf::from("/hab/svc/builder-api/config/service.crt"),
```

Comment:
The default path here is not very meaningful in the sense that we probably won't use these defaults, but that said if we want to put in something it might make more sense to point to "/hab/svc/builder-api/files" directory instead which is where the other keys we gossip around land up in